x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateAddProcessor.java

```diff
@@ -59,14 +59,14 @@ public static Object process(Object unit, Object numberOfUnits, Object timestamp
             }
         }
 
-        if (numberOfUnits instanceof Integer == false) {
-            throw new SqlIllegalArgumentException("An integer is required; received [{}]", numberOfUnits);
+        if (numberOfUnits instanceof Number == false) {
+            throw new SqlIllegalArgumentException("A number is required; received [{}]", numberOfUnits);
         }
 
         if (timestamp instanceof ZonedDateTime == false) {
             throw new SqlIllegalArgumentException("A date/datetime is required; received [{}]", timestamp);
         }
 
-        return datePartField.add(((ZonedDateTime) timestamp).withZoneSameInstant(zoneId), (Integer) numberOfUnits);
+        return datePartField.add(((ZonedDateTime) timestamp).withZoneSameInstant(zoneId), ((Number) numberOfUnits).intValue());
```

Comment:
Wouldn't this mean that in some cases the actual number being passed can be rounded/truncated? If so, should we mention this in the docs?